### PR TITLE
Refactor: lib/jekyll/convertor/markdown.rb - tests: no additions/breaks.

### PR DESCRIPTION
**Pull request by envygeeks**
Reason: #3770
/cc @jekyll/core

### CODE REVIEW

File: lib/jekyll/converters/markdown.rb [hunk 1]
```diff
@@ -1,54 +1,63 @@
...
+      # are not in safe mode.)
+
       def valid_processors
-        %w[
-          rdiscount
-          kramdown
-          redcarpet
-        ] + third_party_processors
+        %W(rdiscount kramdown redcarpet) + \
+          third_party_processors
```
Comment (parkr) [inline lib/jekyll/converters/markdown.rb]: I think this can be on one line. Also, how do `%w` and `%W` differ?

Comment (envygeeks) [inline lib/jekyll/converters/markdown.rb]: They don't differ, it just happened automatically on my end and it's something I should probably remove in favor of the widely used `%w[]` for consistency.

File: lib/jekyll/converters/markdown.rb [hunk 2]
```diff
@@ -56,21 +65,35 @@ def output_ext(ext)
       end
 
       def convert(content)
-        setup
-        @parser.convert(content)
+      setup
```
Comment (parkr) [inline lib/jekyll/converters/markdown.rb]: Let's leave this indented 2 spaces to ensure it's known that it's supposed to be a part of `convert()`.

File: lib/jekyll/converters/markdown.rb [hunk 3]
```diff
@@ -56,21 +65,35 @@ def output_ext(ext)
...
+        @parser.convert(
+          content
+        )
```
Comment (parkr) [inline lib/jekyll/converters/markdown.rb]: I think this can be on one line.

File: lib/jekyll/converters/markdown.rb [hunk 4]
```diff
@@ -56,21 +65,35 @@ def output_ext(ext)
...
       end
 
       private
+      def get_custom_processor
+        md = @config["markdown"]
```
Comment (parkr) [inline lib/jekyll/converters/markdown.rb]: `converter_name`, maybe? `md` confused me -- I thought it was supposed to be the markdown extension, e.g. `".md"` or `".markdown"`.

File: lib/jekyll/converters/markdown.rb [hunk 5]
```diff
@@ -56,21 +65,35 @@ def output_ext(ext)
...
-      # Returns true if the parser name contains only alphanumeric characters
-      # and is defined within Jekyll::Converters::Markdown
-      def allowed_custom_class?(parser_name)
-        parser_name !~ /[^A-Za-z0-9]/ && self.class.constants.include?(parser_name.to_sym)
+      # Returns true if the parser name contains only alphanumeric
+      # characters and is defined within Jekyll::Converters::Markdown
+
+      private
+      def custom_class_allowed?(parser_name)
+        parser_name !~ /[^A-Za-z0-9_]/ && self.class.constants.include?(
```
Comment (parkr) [inline lib/jekyll/converters/markdown.rb]: Let's split this line on the `&&` instead of on the `include?` args.

**Comment by parkr** [comment]
Great! Again, just style things, the substance looks great. Thanks, Jordon!

**Comment by parkr** [comment]
Still want to see this merged, @envygeeks ?

**Comment by envygeeks** [comment]
Yes, I forgot all about this, I'll re-review and it updated for the 3.1 release.

**Pull request merged**